src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -2549,6 +2554,31 @@ private static bool TryEmitMethodCall(object exprObj, object[] paramExprs, ILGen
                 return EmitMethodCall(il, method);
             }
 
+
+            // if call is done into byref method parameters there is no indicators in tree, so grab that from method
+            // current approach is to copy into new list only if there are by ref with by ref parameters,
+            // possible approach to store hit map of small size (possible 256 bit #89) to check if parameter is by ref
+            // https://stackoverflow.com/questions/12658883/what-is-the-maximum-number-of-parameters-that-a-c-sharp-method-can-be-defined-as
+            private static IList<Expression> MakeByRefParameters(MethodCallExpression expr)
+            {
+                IList<Expression> refed = null;
+                var receivingParameters = expr.Method.GetParameters();
+                for (int i = 0; i < expr.Method.GetParameters().Length; i++)
+                {
+                    if (receivingParameters[i].ParameterType.IsByRef)
+                    {
+                        if (refed == null)
+                            refed = new List<Expression>(expr.Arguments);
+                        var passed = expr.Arguments[i] as ParameterExpression;
+                        if (!passed.IsByRef)

```

Comment:
`passed` should be check for `null` via `?.`.
the original argument should be returned.

Then the result may be changed to `IEnimerable<object>`. Is it possible?